components/studio/test/shared/studio-internals/test-all.sh

```diff
@@ -0,0 +1,30 @@
+#!/bin/bash
+
+test_cases=(
+    build
+    metadata
+    hab-version
+)
+
+exit_code=0 
+
+for test_case in "${test_cases[@]}"; do
+  echo "Testing '$test_case'"
+  test/shared/studio-internals/test-studio-"$test_case".sh 
+  if [[ $? -ne 0 ]]; then 
+    echo "FAIL - $test_case"
+    exit_code = 1
+  else
+    echo "PASS - $test_case" 
+  fi 
+done 
+
+
+# This is for expect, as I don't know how to get the result 
+# of a command passed to "send" 
```

Comment:
You're not going to be able to get an exit code from something passed to `send`, since that is really just typing the commands for you.

Instead, I think you'll want to look at some combination of `spawn` and `wait` (e.g., https://stackoverflow.com/a/23632210)

Reply:
It wasn't entirely clear to me if spawn was an option. We want all the test commands to run in the context (inside) of the `spawn`ed studio,  can a subsequent spawn be executed in that context, or will it it always execute on the host (external to the studio)?   This is something I need to look into still 😄